src/actors/player.gd

```diff
@@ -44,6 +47,8 @@ func _ready():
 	cross_hair = get_global_mouse_position()
 	var _i = get_parent().connect("demon_summoned", self, "_on_demon_summoned")
 	var _x = get_parent().connect("repair_tick", self, "_on_repair_tick")
+	enemy_node = get_tree().get_current_scene().get_node("EnemyNode")
```

Comment:
The fact that the `Player` needs access to the `EnemyNode` and the `MeleeMob` needs access to the `Player` **and** the `EnemyNode` feels like a code smell to me. Nothing to do in this PR, but there is something here that should be handled via signals.

Also the fact that _implicit_ in both of these references is that the current scene that the Enemy or Player is _in_ **must** have an EnemyNode is another bit of "out of scene" coupling we will need to figure out how to best address.

Reply:
removed enemy node because we can just create a group called "movable" that all of these entities are part of and check if the collided entity is part of that group